src/firefly/js/visualize/region/Region.js

```diff
@@ -0,0 +1,291 @@
+ */
+
+/**
+ * This Object contains the specifications of the DS9 region
+ */
+
+import Enum from 'enum';
+
+export const RegionType = new Enum(['circle', 'annulus', 'ellipse', 'ellipseannulus', 'box',
+                                    'boxannulus', 'line', 'point', 'polygon', 'text', 'undefined']);
+
+export const RegionCsys = new Enum(['PHYSICAL', 'FK4', 'B1950', 'FK5', 'J2000',
+                                    'IMAGE', 'ECLIPTIC', 'GALACTIC',
+                                    'ICRS', 'AMPLIFIER', 'LINEAR', 'DETECTOR', 'UNDEFINED']);
+
+export const RegionValueUnit = new Enum(['CONTEXT', 'DEGREE', 'RADIUS', 'ARCMIN', 'ARCSEC',
+                                         'SCREEN_PIXEL', 'IMAGE_PIXEL']);
+
+export const RegionPointType = new Enum(['circle', 'box', 'cross', 'diamond', 'x',
+                                         'arrow', 'boxcircle', 'undefined']);
+
+
+/**
+ *
+ * @param wpAry WorldPt array, items contained: line, annulus, ellipse, box: 1, line: 2, polygon: >= 3
+ * @param radiusAry RegionValue array, items contained: annulus: at least 2 radii.
+ * @param dimensionAry RegionDimension array, each item contains w, h for box and box annulus
+ *                                            or 2 radii for ellipse and ellipse annulus
+ * @param angle RegionValue
+ * @param type  RegionType
+ * @param options   RegionOptions, properties of region
+ * @param highlighted bool
+ * @constructor
+ */
+export var Region =
+    ( {wpAry = null,
+      radiusAry = null,
+      dimensionAry = null,
+      angle = null,
+      type = RegionType.undefined,
+      options = null,
+      highlighted = false}
+    ) => ({type, wpAry, radiusAry, dimensionAry, angle, options, highlighted});
+
+/**
+ *
+ * @param color     string, 'white', 'black', 'red', 'green', 'blue', 'cyan', 'magenta', 'yellow'
+ * @param text      string
+ * @param font      RegionFont
+ * @param pointType     RegionPointType
+ * @param pointSize     number
+ * @param editable      bool
+ * @param movable       bool
+ * @param rotatable     bool
+ * @param highlightable bool
+ * @param deletable     bool
+ * @param fixedSize     bool
+ * @param include       bool
+ * @param lineWidth     number
+ * @param offsetX       number
+ * @param offsetY       number
+ * @param message       string, containing parsing message
+ * @constructor
+ */
+
+export var RegionOptions =
+    ( {color = 'blue',
+        text  = null,
+        font  = null,
+        pointType = null,
+        pointSize =  0,
+        editable  =  true,
+        movable   =  true,
+        rotatable =  true,
+        highlightable =  true,
+        deletable = true,
+        fixedSize = false,
+        include   = true,
+        lineWidth = 0,
+        offsetX   = 0,
+        offsetY   = 0,
+        message  = null}
+    ) => ({color, text, font, pointType, pointSize, editable, movable, rotatable, highlightable,
+        deletable, fixedSize, include, lineWidth, offsetX, offsetY, message});
+
```

Comment:
I have similar feeling about unnecessary keys here, but since they mostly default to true it is harder.
 Add a line note